### PR TITLE
CVSL-227: Split standard and additional conditions to AP and PSS standard conditions

### PR DESCRIPTION
**Pull request by smcveigh941**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/LicenceControllerTest.kt [hunk 1]
```diff
@@ -483,8 +483,10 @@ class LicenceControllerTest {
       probationLduCode = "LDU1",
       dateCreated = LocalDateTime.now(),
       createdByUsername = "X12345",
-      standardConditions = someStandardConditions,
-      additionalConditions = someAdditionalConditions,
+      standardLicenceConditions = someStandardConditions,
+      standardPssConditions = someStandardConditions,
+      additionalLicenceConditions = someAdditionalConditions,
+      additionalPssConditions = someAdditionalConditions,
```
Comment (timharrison-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/LicenceControllerTest.kt:489]: Will this result in the same type of standard and additional conditions on a licence? ie. not separated into different types?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/LicenceControllerTest.kt [hunk 2]
```diff
@@ -516,7 +518,8 @@ class LicenceControllerTest {
       comTelephone = "0116 2788777",
       probationAreaCode = "N01",
       probationLduCode = "LDU1",
-      standardConditions = someStandardConditions,
+      standardLicenceConditions = someStandardConditions,
+      standardPssConditions = someStandardConditions
     )
```
Comment (timharrison-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/LicenceControllerTest.kt:523]: As above with this one too?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/AdditionalConditionsRequest.kt [hunk 3]
```diff
@@ -8,4 +8,8 @@ data class AdditionalConditionsRequest(
   @Schema(description = "The list of additional conditions")
   @field:NotNull
   val additionalConditions: List<AdditionalCondition>,
+
+  @Schema(description = "The type of additional condition, either licence or post sentence supervision", allowableValues = ["AP", "PSS"])
+  @field:NotNull
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/AdditionalConditionsRequest.kt:13]: Are we expecting there to be different standard and additional conditions by licence type? I don't know the answer to that, but just a thought. We might end up with convoluted config if there is no re-use?

**Review by timharrison-moj — APPROVED** [review]
LGTM. A nice and neat way to separate out the different condition types without a lot of complexity. Couple of random thoughts added.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/AdditionalConditionsRequest.kt (hunk 3 above)
Comment (smcveigh941) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/AdditionalConditionsRequest.kt:13]: My understanding is that there is one list of AP standard/additional conditions, and a list of PSS standard/additional conditions.

AP and PSS licences would have the respective condition types, and AP+PSS would have both

File: src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/LicenceControllerTest.kt (hunk 1 above)
Comment (smcveigh941) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/LicenceControllerTest.kt:489]: The licence object and the createLicence request object are agnostic of the `conditionType` field. It only splits them out into separate condition collections as they are here

**Pull request merged**